src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -647,42 +648,182 @@ public void Fault(RequestChannel channel)
 
-                        this.TryCompleteHttpRequest(_httpRequestMessage);
-                        return replyMessage;
+                                _httpRequestMessage.Headers.Referrer = new Uri(value);
+                            }
+                            else if (string.Compare(name, "transfer-encoding", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                if (value.ToUpperInvariant().IndexOf("CHUNKED", StringComparison.OrdinalIgnoreCase) != -1)
+                                {
+                                    _httpRequestMessage.Headers.TransferEncodingChunked = true;
+                                }
+                                else
+                                {
+                                    _httpRequestMessage.Headers.TransferEncoding.TryParseAdd(value);
+                                }
+                            }
+                            else if (string.Compare(name, "user-agent", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                _httpRequestMessage.Headers.UserAgent.Add(ProductInfoHeaderValue.Parse(value));
+                            }
+                            else if (string.Compare(name, "if-modified-since", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                DateTimeOffset modifiedSinceDate;
+                                if (DateTimeOffset.TryParse(value, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AllowWhiteSpaces | DateTimeStyles.AssumeLocal, out modifiedSinceDate))
+                                {
+                                    _httpRequestMessage.Headers.IfModifiedSince = modifiedSinceDate;
+                                }
+                                else
+                                {
+                                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                                        new ProtocolException(SR.Format(SR.HttpIfModifiedSinceParseError, value)));
+                                }
+                            }
+                            else if (string.Compare(name, "date", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // this will be taken care of by System.Net when we make the request
+                            }
+                            else if (string.Compare(name, "proxy-connection", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                throw ExceptionHelper.PlatformNotSupported("proxy-connection");
+                            }
+                            else if (string.Compare(name, "range", StringComparison.OrdinalIgnoreCase) == 0)
+                            {
+                                // we don't support ranges in v1.
```

Comment:
I'll fix the comment. This was copied verbatim from the existing code which came from the desktop. It would seem reasonable to throw an exception here as it's a header where a customer has explicitly provided it and we're silently dropping it instead of basically saying that makes no sense. But throwing an exception here would be different behavior from the desktop. This header is a strange one, as the idea of the range header is to be able to pick up where you left off if a connection dropped, or to make multiple concurrent requests to improve bandwidth, but that doesn't make sense for SOAP. For example, making the exact same request twice to an endpoint using message security (I know we don't support that yet, but we do on desktop) can result in a different response, and piecing those pieces of response together isn't possible. Also it would require the same service request to be executed twice. The only time range makes sense is with a static resource. The ByteStreamMessageEncoder is the only way this could even make sense, and we haven't included that.